sashimi/camera.py

```diff
@@ -124,158 +128,42 @@ def run(self):
-
-class MockCameraProcess(Process):
-    def __init__(
-        self,
-        experiment_start_event,
-        stop_event,
-        camera_id=0,
-        max_queue_size=1200,
-        n_fps_frames=20,
-    ):
-        super().__init__()
-        self.experiment_start_event = experiment_start_event
-        self.stop_event = stop_event
-        self.image_queue = ArrayQueue(max_mbytes=max_queue_size)
-        self.parameter_queue = Queue()
-        self.camera_id = camera_id
-        self.camera = None
-        self.parameters = CamParameters()
-        self.new_parameters = copy(self.parameters)
-        self.camera_status_queue = Queue()
-        self.triggered_frame_rate_queue = Queue()
-        self.framerate_rec = FramerateRecorder(n_fps_frames=n_fps_frames)
-
-    def cast_parameters(self):
-        params = self.parameters
-        params.subarray = list(params.subarray)
-        return params
-
-    def initialize_camera(self):
-        self.camera_status_queue.put(self.cast_parameters())
-
-    def pause_loop(self):
-        while not self.stop_event.is_set():
-            try:
-                self.new_parameters = self.parameter_queue.get(timeout=0.001)
-                if self.new_parameters != self.parameters:
-                    break
-            except Empty:
-                pass
-
-    def run(self):
-        self.initialize_camera()
-        self.run_camera()
-
-    def run_camera(self):
-        while not self.stop_event.is_set():
-            self.parameters = self.new_parameters
-            self.apply_parameters()
-            self.camera_status_queue.put(self.cast_parameters())
-            if self.parameters.camera_mode == CameraMode.PAUSED:
-                self.pause_loop()
-            else:
-                self.camera_loop()
-
-    def camera_loop(self):
-        while not self.stop_event.is_set():
-            time.sleep(0.001)
-            is_there_frame = np.random.random_sample(1)
-            if is_there_frame < (1 / self.parameters.exposure_time):
-                frame = np.random.randint(
-                    0, 65534, size=self.parameters.frame_shape, dtype=np.uint16
-                )
-                self.image_queue.put(frame)
-                self.update_framerate()
-            try:
-                self.new_parameters = self.parameter_queue.get(timeout=0.001)
-                if self.parameters.camera_mode == CameraMode.ABORT or (
-                    self.new_parameters != self.parameters
-                ):
-                    break
-            except Empty:
-                pass
+    def update_parameters(self):
+        self.parameters = self.new_parameters
+        for attribute in ["exposure_time", "binning", "roi", "trigger_mode"]:
+            setattr(self.camera, attribute, getattr(self.parameters, attribute))
+        # the following is to match internals of the camera as some cameras are restricted (e.g. only multiples of 4)
+        self.parameters.frame_shape = self.camera.frame_shape
```

Comment:
I think this is also the case for exposure. But these parameters are (right now) not sent to the main process, right?

Reply:
In 'camera_gui' there is a function to show the user the current shape of the frames:

```python    
    def update_roi_info(self, width, height):
        self.lbl_roi.setText(
            "Current frame dimensions are:\nHeight: {}\nWidth: {}".format(
                int(height), int(width)
            )
        )
```

Now, this function is actually wrong because it is diplaying height and width which are some legacy parameters from previous versions. It should be changed so that it takes intead the `frame_shape` tuple and displays it. This can be done in this PR but also can be done in a further one dealing with #87 .

Reply:
Ok!